lib/ask/runtime/session_mode.ex

```diff
@@ -11,9 +11,13 @@ defmodule Ask.Runtime.SessionModeProvider do
   defp mode_provider("sms"), do: SMSMode
   defp mode_provider("ivr"), do: IVRMode
   defp mode_provider("mobileweb"), do: MobileWebMode
-  defp mode_provider("sms_simulator"), do: SMSSimulatorMode
 
   def new(nil, _channel, _retries), do: nil
+
+  def new(_mode, %Ask.Runtime.SimulatorChannel{} = channel, retries) do
```

Comment:
The reason we don't do this
```suggestion
  def new("sms", %Ask.Runtime.SimulatorChannel{} = channel, retries) do
```
is defensive programming?

Potentially (we don't expect it) we'd be using `SMSSimulatorMode` under unsupported mode scenarios

Reply:
You are right. There is no reason for doing it as it is